camera_manager_plugin: fix mavlink messages for camera info and video stream information

MaEtUgR opened the pull request:
### Problem

Can't update MAVLink in PX4 anymore because of
https://github.com/mavlink/mavlink/pull/2121
https://github.com/mavlink/mavlink/pull/2127
which are not breaking changes but because of the way they are invoked here end up breaking the build
which makes CI in https://github.com/PX4/PX4-Autopilot/pull/23490 fail.

### Solution
Like proposed by @dagar this time I fix it with the struct call instead of the direct pack_chan() calls. So it should hold longer.

Please check if I did any mistake with these string, component ID and MAVLink channel things. What was `MAVLINK_COMM_1` used for?

MaEtUgR committed:
camera_manager_plugin: fix mavlink messages for camera info and video stream information

This time with structs instead of the direct pack_chan() calls that might break again quickly.

dagar commented on src/gazebo_camera_manager_plugin.cpp:
```diff
@@ -614,34 +614,30 @@ void CameraManagerPlugin::_handle_camera_info(const mavlink_message_t *pMsg, str
+                                | CAMERA_CAP_FLAGS_HAS_VIDEO_STREAM;;
+    camera_information.cam_definition_version = 0;
+    std::strcpy(camera_information.cam_definition_uri, "");
+    camera_information.gimbal_device_id = 0;
```
You could skip all these fields you're not actually setting.

MaEtUgR replied:
ok, I left away all the zero assignments in 357396f8c048a8ceadb14cfc6409817b2079d15b

hamishwillee commented on src/gazebo_camera_manager_plugin.cpp:
```diff
@@ -720,25 +714,22 @@ void CameraManagerPlugin::_handle_request_video_stream_information(const mavlink
+    video_stream_information.hfov = 90; // made up
+    std::strcpy(video_stream_information.name, "Visual Spectrum");
+    std::strcpy(video_stream_information.uri, _videoURI.c_str());
+
```
Note, there is a new encoding field, which ideally you should set to [VIDEO_STREAM_ENCODING_H264](https://mavlink.io/en/messages/common.html#VIDEO_STREAM_ENCODING_H264). If you don't it won't matter because the default value 0 means "don't know" and the recipient should detect. This is to reduce effort on recipient as cameras start to encode in H265.

![image](https://github.com/user-attachments/assets/d9b9b3d0-bede-4e83-88f8-755c3d7dd140)

MaEtUgR replied:
You're probably right, didn't even check what should be set in the extension 😇
I added it in 357396f8c048a8ceadb14cfc6409817b2079d15b

This will probably make CI fail again because the extension does not yet exist on PX4 main until https://github.com/PX4/PX4-Autopilot/pull/23490 is merged...

julianoes commented on src/gazebo_camera_manager_plugin.cpp:
```diff
@@ -614,34 +614,30 @@ void CameraManagerPlugin::_handle_camera_info(const mavlink_message_t *pMsg, str
+
+    mavlink_camera_information_t camera_information{};
+    camera_information.time_boot_ms = 0;
+    std::strcpy((char *)camera_information.vendor_name, "PX4.io");
```
In general, I'd advocate to use `std::strncpy()` instead and add the max of the destination (here the length in the MAVLink message).

MaEtUgR replied:
Thanks for having a look! Those are all short fixed-size strings but you are completely correct, I learned something and adjusted them in 357396f8c048a8ceadb14cfc6409817b2079d15b

MaEtUgR committed:
camera_manager_plugin: skip all these fields that are not set, use stncpy with destination size, set H264 encoding

MaEtUgR committed:
camera_manager_plugin: bring back MAVLINK_COMM_1 channel

I don't understand if and why it's necessary but found a way to do it with the struct encode method.

dagar commented on src/gazebo_camera_manager_plugin.cpp:
```diff
@@ -720,25 +708,22 @@ void CameraManagerPlugin::_handle_request_video_stream_information(const mavlink
+    video_stream_information.hfov = 90; // made up
+    std::strncpy(video_stream_information.name, "Visual Spectrum", sizeof(video_stream_information.name));
+    std::strncpy(video_stream_information.uri, _videoURI.c_str(), sizeof(video_stream_information.uri));
+    video_stream_information.encoding = VIDEO_STREAM_ENCODING_H264;
```
```suggestion
```

dagar committed:
Update src/gazebo_camera_manager_plugin.cpp
